### ISSUE TITLE
pay plugin and single quote / label / description issue

### ISSUE THREAD

**Issue by rsbondi** [ElementsProject/lightning#2597]
this may or may not be related to #2010 but I discovered the following

`lightning-cli pay lnbcrt... null "ain't right"` it crashes

```
pay: plugins/libplugin.c:166: vprintf_json: Assertion `n % 2 == 0' failed.
pay: FATAL SIGNAL 6 (version v0.7.0-276-g62e1423)
0x559ea87ffd4e send_backtrace
	common/daemon.c:40
0x559ea87ffdf4 crashdump
	common/daemon.c:53
0x7fa59d10df1f ???
	???:0
0x7fa59d10de97 ???
	???:0
0x7fa59d10f800 ???
	???:0
0x7fa59d0ff399 ???
	???:0
0x7fa59d0ff411 ???
	???:0
0x559ea87f4225 vprintf_json
	plugins/libplugin.c:166
0x559ea87f4d58 send_outreq_
	plugins/libplugin.c:415
0x559ea87f1315 getroute_done
	plugins/pay.c:514
0x559ea87f4b6b handle_rpc_reply
	plugins/libplugin.c:381
0x559ea87f57bd plugin_main
	plugins/libplugin.c:603
0x559ea87f3b56 main
	plugins/pay.c:1216
0x7fa59d0f0b96 ???
	???:0
0x559ea87ef8f9 ???
	???:0
0xffffffffffffffff ???
	???:0
pay: FATAL SIGNAL 11 (version v0.7.0-276-g62e1423)
0x559ea87ffd4e send_backtrace
	common/daemon.c:40
0x559ea87ffdf4 crashdump
	common/daemon.c:53
0x7fa59d10df1f ???
	???:0
0x0 ???
	???:0

```

`lightning-cli pay lnbcrt... null "ain't workin'"` (n % 2 == 0) does not crash but gives log of

```
pay: JSON reply without numeric id '{ "jsonrpc": "2.0", "id" : null, "error" : { "code" : -32600, "message" : "Invalid token in json input" } }
```

So first of all, I am a bit confused about what is the intention of `label`.  The response to `pay` shows `description` and `label` with the same value, this lead me to the assumption that they had similar intention.  If the `bolt11` has a `description`, it is totally ignored if you omit the optional parameters.  But I figured maybe it is up to the client to parse, so that is what I started doing, parse the ` bolt11` and throw the description into  the UI to get submitted as the `label` of the `pay` command and allowing it to be edited if desired.

Apparently,  `label` in `pay` does not have the same restriction of being unique like with `invoice` so it seems to further strengthen my original assumption, I noticed deprecation warning in the code so `label` is the new `description`?

Second, this issue seems to stem from trying to fix invalid JSON, or at least it appears that way from this code snippet from `libplugin.c`

```c
	for (n = 0, p = strchr(json, '\''); p; p = strchr(json, '\'')) {
		*p = '"';
		n++;
	}
	/* Don't put stray single-ticks in like this comment does! */
	assert(n % 2 == 0);
```
which if my assumption is correct, then this seems like a bad idea.

So if the intention for `label` is to allow user entered information, this needs to be handled to allow `'` and would be nice to use the `description` from the `bolt11` if `label` is omitted.  If the intention is to have the client generate an identifier of sorts, then it would be nice to clarify in the help and or man page.

**Comment by rsbondi** [comment]
This appears to be fixed with #2705 pending merge

**Comment by rsbondi** [comment]
this is now working with latest release
